vortex-datafusion/src/eval.rs

```diff
@@ -14,34 +14,35 @@ impl ExpressionEvaluator {
     pub fn eval(array: Array, expr: &Expr) -> VortexResult<Array> {
         match expr {
             Expr::BinaryExpr(expr) => {
-                let lhs = expr.left.as_ref();
-                let rhs = expr.right.as_ref();
+                let lhs = ExpressionEvaluator::eval(array.clone(), expr.left.as_ref())?;
+                let rhs = ExpressionEvaluator::eval(array, expr.right.as_ref())?;
 
                 // TODO(adamg): turn and/or into more general compute functions
                 match expr.op {
                     DFOperator::And => {
-                        let lhs = ExpressionEvaluator::eval(array.clone(), lhs)?.into_bool()?;
-                        let rhs = ExpressionEvaluator::eval(array, rhs)?.into_bool()?;
+                        let lhs = lhs.into_bool()?;
+                        let rhs = rhs.into_bool()?;
                         let buffer = &lhs.boolean_buffer() & &rhs.boolean_buffer();
                         Ok(BoolArray::from(buffer).into_array())
                     }
                     DFOperator::Or => {
-                        let lhs = ExpressionEvaluator::eval(array.clone(), lhs)?.into_bool()?;
-                        let rhs = ExpressionEvaluator::eval(array.clone(), rhs)?.into_bool()?;
+                        let lhs = lhs.into_bool()?;
+                        let rhs = rhs.into_bool()?;
                         let buffer = &lhs.boolean_buffer() | &rhs.boolean_buffer();
                         Ok(BoolArray::from(buffer).into_array())
                     }
-                    DFOperator::Eq => {
-                        let lhs = ExpressionEvaluator::eval(array.clone(), lhs)?;
-                        let rhs = ExpressionEvaluator::eval(array.clone(), rhs)?;
-                        compare(&lhs, &rhs, Operator::Eq)
-                    }
+                    DFOperator::Eq => compare(&lhs, &rhs, Operator::Eq),
```

Comment:
Can we define a `TryFrom<DFOperator> for Operator`? And then use that to check for support?

Reply:
`And` and `Or` don't really fit into how `Operator` is designed right now with functions like `to_predicate` or even `inverse`